clang/lib/Frontend/CompilerInvocation.cpp

```diff
@@ -4246,6 +4246,34 @@ void CompilerInvocationBase::GenerateLangArgs(const LangOptions &Opts,
     GenerateArg(Consumer, OPT_frandomize_layout_seed_EQ, Opts.RandstructSeed);
 }
 
+static bool parseFeatureAvailability(const Arg *A, LangOptions &Opts,
+                                     DiagnosticsEngine &Diags) {
+  llvm::StringRef ArgRef(A->getValue());
+  auto Split = ArgRef.split(':');
+  llvm::StringRef Feature = Split.first;
+  llvm::StringRef StateStr = Split.second;
+
+  if (Feature.empty() || StateStr.empty()) {
+    Diags.Report(clang::diag::err_feature_availability_flag_invalid_value)
+        << ArgRef;
+    return false;
+  }
+
+  FeatureAvailKind State;
+  if (StateStr == "on") {
+    State = FeatureAvailKind::Available;
+  } else if (StateStr == "off") {
+    State = FeatureAvailKind::Unavailable;
+  } else {
+    Diags.Report(clang::diag::err_feature_availability_flag_invalid_value)
```

Comment:
I'd suggest allowing a few more spellings, so as to make this easier to drive from CMake: `on, On, off, Off, 0, 1`